### PR TITLE
build: migrate to pnpm

### PR DESCRIPTION
**Pull request by Dafnik**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/analogjs/analog/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [X] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] vite-plugin-angular
- [ ] vite-plugin-nitro
- [ ] astro-angular
- [] create-analog
- [ ] router
- [ ] platform
- [ ] content
- [ ] nx-plugin
- [ ] trpc

## What is the current behavior?

Currently `yarn` is used for dependency management.

Build upon #192

## What is the new behavior?

Migrates project to `pnpm`. 

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

## Other information
This pr also works in the vite-ecosystem-ci

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -33,20 +33,18 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
+      - uses: dafnik/setup-node-pnpm@v1
```
Comment (brandonroberts) [inline .github/workflows/ci.yml:36]: Why do we need a custom actions library here? I had pnpm working in the other PR with the standard package.

File: packages/nx-plugin/src/generators/app/lib/initialize-analog-workspace.ts [hunk 2]
```diff
@@ -55,7 +55,7 @@ export async function initializeAngularWorkspace(
         tree,
         {},
         {
-          '@nrwl/devkit': V15_NRWL_DEVKIT,
+          '@nx/devkit': V15_NRWL_DEVKIT,
```
Comment (brandonroberts) [inline packages/nx-plugin/src/generators/app/lib/initialize-analog-workspace.ts:58]: Nx 15.x should still be @nrwl

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Dafnik) [reply .github/workflows/ci.yml:36]: Yeah, you are right. I was just testing a few things and forgot to revert it back.

File: packages/nx-plugin/src/generators/app/lib/initialize-analog-workspace.ts (hunk 2 above)
Comment (Dafnik) [reply packages/nx-plugin/src/generators/app/lib/initialize-analog-workspace.ts]: My mistake. You should be fixed.

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**